### PR TITLE
Add a bit more test coverage for datacube ext

### LINKED ISSUE #324: Tests and documentation for Datacube extension
As part of #309 the Datacube extension was implemented (closing #30). There's testing and documentation left to do on this extension that should go in before the 1.0 release.

### PR DESCRIPTION
**Pull request by jsignell**
**Related Issue(s):**

- Closes #324

**Description:**

There is already a fair bit of documentation in there, but I improved the test coverage a bit. This is a very good example of an extension class that could/should be substantially simplified in the 2.0 release

**PR Checklist:**

- [x] `pre-commit` hooks pass locally
- [x] Tests pass (run `scripts/test`)
- [x] Documentation has been updated to reflect changes, if applicable
- [x] This PR maintains or improves overall codebase code coverage.
- [ ] Changes are added to the [CHANGELOG](https://github.com/stac-utils/pystac/blob/main/CHANGELOG.md). See [the docs](https://pystac.readthedocs.io/en/latest/contributing.html#changelog) for information about adding to the changelog.

### CODE REVIEW

**Comment by jsignell** [comment]
Ok so patch coverage is low, but a significant improvement :shrug: seems to me like it would be ok to override that check, but I could probably also coverage up high enough with another hour or so of work. Just let me know what makes more sense to you @gadomski

**Comment by gadomski** [comment]
Nah, let's ignore the patch ❌, it seems like too much work to test out all the extension 🍝.

**Review by jpolchlo — APPROVED** [review]
Minor nits, but LGTM

File: tests/extensions/test_datacube.py [hunk 1]
```diff
@@ -1,102 +1,269 @@
...
+    dim = horizontal_dimension
+    assert dim.values is None
+
+    dim.values = [0, 180]
+    assert dim.values == [0, 180]
+    assert dim.properties["values"] == [0, 180]
+
+    dim.clear_step()
+    assert dim.step is None
+    assert "step" not in dim.properties
```
[the remark this answers is not in the document]
Comment (jsignell) [reply tests/extensions/test_datacube.py:72]: I think this is a bad copy and paste left over from a different organization pattern that I had at one point. Good catch.

**Comment by jsignell** [comment]
@gadomski if this one merges it is maybe a little concerning because it was failing coverage. It is fine in this case as mentioned above, but just wanted to flag this.

**Comment by gadomski** [comment]
> if this one merges it is maybe a little concerning because it was failing coverage.

`codecov/patch` is an optional check, so its not surprising it merged. I'm a little hesitant to make it required, since patch coverage can be so fiddly to get right.

**Comment by jsignell** [comment]
nice! Ok I just hadn't noticed it was explicitly optional

**Pull request merged**